shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -122,7 +154,21 @@ private case class CloseableColumnBatchIterator(iter: Iterator[ColumnarBatch]) e
 class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
   override def supportsColumnarInput(schema: Seq[Attribute]): Boolean = true
 
-  override def supportsColumnarOutput(schema: StructType): Boolean = true
+  override def supportsColumnarOutput(schema: StructType): Boolean = schema.fields.forall(f =>
+    isSupportedByCudf(f.dataType) || isSupportedBySparkColumnar(f.dataType))
```

Comment:
@jlowe The reason we are checking for both is because if either one supports the columnar output we should call the `convertCachedBatchToColumnarBatch` otherwise we should call `convertCachedBatchToInternalRow`.
In case a type is supported on the CPU and not the GPU, we will call the appropriate place to decode it because I will call the `isSupportedByCudf` or `isSupportedBySparkColumnar` before I decode it. e.g. In case of ArrayType, we will encode it on GPU and when reading it back I will make the decision of where to decode it by calling the `isSupportedByCudf` again. Is this confusing you because currently I am not making that call in the `convertCachedBatchToColumnar`? 

@revans2 yes, that is the long term goal but in the interim, I will need this check in place don't I? Also, the fact that we replace the InMemoryTableScan only means that we are using the `ParquetCachedBatchSerializer` which could mean that the cachedbatch that we are about to read could be read on the CPU if we don't support everything on the GPU. Please correct me if I am wrong

Reply:
> Also, the fact that we replace the InMemoryTableScan only means that we are using the ParquetCachedBatchSerializer

No because there is also an implicit check for all operators where it verifies that the types being used are supported. The fact that we replaced it means the plugin is enabled and all of the types are things supported by the GPU.

Reply:
OK, but `ParquetCachedBatchSerializer` can still be used even if we haven't replaced the `InMemoryTableScan` that means we should still make sure we can support everything on the GPU before blindly pushing it. At least that's how I understood this

Reply:
nevermind, I was wrong. The fact that we haven't replaced the InMemoryTableScan means that we can't support everything on the GPU so only CPU check should be made.